### PR TITLE
Mldb 1190 segfault sqlexpr jseval biz

### PR DESCRIPTION
**Pull request by guyd**
When running the jseval query below a seg fault was occurring after several calls.  The problem is that we were performing a static_cast on an object of the wrong type.  This happens when a function calls BoundSqlExpression.constantValue() on an SQL expression bound with a ReadThroughBindingContext.  This is the case of jseval and regexp functions when called from within an sql.expression.

### CODE REVIEW

**Comment by guyd** [comment]
@jeremybarnes This is the problem I was describing earlier this morning.

**Comment by steadtler** [comment]
I guess you meant static_cast in your description, not const_cast :)

File: sql/binding_contexts.cc [hunk 1]
```diff
@@ -32,8 +32,19 @@ rebind(BoundSqlExpression expr)
                      ExpressionValue & storage)
         -> const ExpressionValue &
         {
-            auto & row = static_cast<const RowContext &>(context);
```
Comment (steadtler) [inline sql/binding_contexts.cc]: Could / should this be a virtual method instead?

**Comment by steadtler** [comment]
I'll let you decide for the virtual method, take it as a suggestions, +1 when you're decided.

**Comment by guyd** [comment]
regarding the description, yes my bad, I meant static_cast.

**Comment by jeremybarnes** [comment]
Hmmm, this smells like a design issue we're plastering over.

**Comment by guyd** [comment]
It not only smells, it is a plaster!  I will revisit once I'm done with my current task.

**Comment by guyd** [comment]
I revisited the code.  This is more aligned with the design.  Let me know what you think.

**Comment by steadtler** [comment]
Well that is much better heh! + 1

**Comment by jeremybarnes** [comment]
Indeed... much much better :)

**Pull request merged**